guides/v2.1/cloud/project/project-conf-files_magento-app.md

```diff
@@ -206,59 +207,94 @@ hooks:
+
+```yaml
+hooks:
+    # We run build hooks before your application has been packaged.
+    build: |
+        php ./vendor/bin/ece-tools build:generate
+        # php /path/to/your/script
+        php ./vendor/bin/ece-tools build:transfer
+```
+
+-  `build:generate`—applies patches, validates configuration, generates DI, and generates static content if SCD is enabled for build phase.
+-  `build:transfer`—transfers generated code and static content to the final destination.
+
 The commands run from the application (`/app`) directory. You can use the `cd` command to change the directory. The hooks fail if the final command in them fails. To cause them to fail on the first failed command, add `set -e` to the beginning of the hook.
 
 #### To compile SASS files using grunt:
 
 ```yaml
 dependencies:
     ruby:
-        sass: "3.4.7"
+      sass: "3.4.7"
     nodejs:
-        grunt-cli: "~0.1.13"
+      grunt-cli: "~0.1.13"
 
 hooks:
     build: |
         cd public/profiles/project_name/themes/custom/theme_name
         npm install
         grunt
         cd
-        php ./bin/magento magento-cloud:build
+        php ./vendor/bin/ece-tools
 ```
 
 You must compile SASS files using `grunt` before static content deployment, which happens during the build. Place the `grunt` command before the `build` command.
 
 ### `crons`
-Describes processes that are triggered on a schedule. We recommend you run cron as the [Magento file system owner]({{ page.baseurl }}/cloud/before/before-workspace-file-sys-owner.html). Do _not_ run cron as `root` or as the web server user.
+Describes processes that are triggered on a schedule. We recommend you run `cron` as the [Magento file system owner]({{page.baseurl}}/cloud/before/before-workspace-file-sys-owner.html). Do _not_ run cron as `root` or as the web server user.
+
+`crons` support the following:
 
-More information about crons:
+-  `spec`—The cron specification. For Starter environments and Pro Integration environments, the minimum interval is once per five minutes and once per one minute in Pro Staging and Production environments. You need to complete [additional configurations]({{ page.baseurl }}/cloud/configure/setup-cron-jobs.html#add-cron) for crons in those environments.
+-  `cmd`—The command to execute.
 
-`crons` supports the following:
+A Cron job is well suited for the following tasks:
 
-*	`spec`: The cron specification. The minimum interval is once per 5 minutes.
-*	`cmd`: The command to execute.
+-  They need to happen on a fixed schedule, not continually.
+-  The task itself is not especially long, as a running cron job will block a new deployment.
+-  Or it is long, but can be easily divided into many small queued tasks.
+-  A delay between when a task is registered and when it actually happens is acceptable.
 
 A sample Magento cron job follows:
 
 ```yaml
 crons:
   cronrun:
-    spec: "*/5 * * * *"
-    cmd: "php bin/magento cron:run"
+      spec: "*/5 * * * *"
```

Comment:
Update this to fit latest template: https://github.com/magento/magento-cloud/blob/master/.magento.app.yaml#L89